pkg/balancer-js/src/pool-utils/swapper.ts

```diff
@@ -0,0 +1,149 @@
+import { Signer } from '@ethersproject/abstract-signer';
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+import { AddressZero, Zero } from '@ethersproject/constants';
+import { ContractTransaction, PayableOverrides } from '@ethersproject/contracts';
+import { Provider } from '@ethersproject/providers';
+import invariant from 'tiny-invariant';
+import { Vault, Vault__factory, WETH, WETH__factory } from '@balancer-labs/typechain';
+import { BatchSwapStep, FundManagement, SingleSwap, SwapKind } from '../types';
+
+/**
+ * Converts a batch swap which interacts with only a single pool to an equivalent simple swap
+ * @param swap - a SingleSwap object describing the pool to perform the swap against.
+ * @returns an equivalent batch swap
+ */
+export const convertSingleToBatchSwap = (
+  swap: SingleSwap,
+  limit: BigNumberish
+): { kind: SwapKind; swaps: BatchSwapStep[]; assets: string[]; limits: BigNumber[] } => {
+  const assets = [swap.assetIn, swap.assetOut];
+  const swaps = [
+    {
+      poolId: swap.poolId,
+      assetInIndex: assets.indexOf(swap.assetIn),
+      assetOutIndex: assets.indexOf(swap.assetOut),
+      amount: swap.amount,
+      userData: swap.userData,
+    },
+  ];
+  const limits = swap.kind === SwapKind.GivenIn ? [BigNumber.from(limit), Zero] : [Zero, BigNumber.from(limit).mul(-1)];
+  return { kind: swap.kind, swaps, assets, limits };
+};
+
+/**
+ * Converts a batch swap which interacts with only a single pool to an equivalent simple swap
+ * @param swaps - a SingleSwap object describing the pool to perform the swap against.
+ * @param funds - a FundManagement object describing where funds are drawn from and then sent after the trade.
+ * @param limit - an array of the maximum allowable inflows to the vault for each token. Negative values describe minimum outflows.
+ * @param deadline - the unix timestamp after which this trade is no longer valid
+ * @returns an equivalent SingleSwap object
+ */
+export const convertBatchToSingleSwap = (
+  kind: SwapKind,
+  swaps: BatchSwapStep[],
+  assets: string[],
+  limits: BigNumberish[]
+): { swap: SingleSwap; limit: BigNumber } => {
+  invariant(swaps.length === 1, 'BatchSwap not convertable to SingleSwap');
+  invariant(assets.length === 2, 'BatchSwap not convertable to SingleSwap');
+  invariant(limits.length === 2, 'BatchSwap not convertable to SingleSwap');
+  const { poolId, assetInIndex, assetOutIndex, amount, userData } = swaps[0];
+
+  const swap = {
+    poolId,
+    kind,
+    assetIn: assets[assetInIndex],
+    assetOut: assets[assetOutIndex],
+    amount,
+    userData,
+  };
+  const limit =
+    swap.kind === SwapKind.GivenIn
+      ? BigNumber.from(limits[assetInIndex])
+      : BigNumber.from(limits[assetOutIndex]).mul(-1);
```

Comment:
Same here, on a given in swap you care about the limit placed on the amount out, not in.